docs/RationaleHW.md

```diff
@@ -142,6 +142,69 @@ notable differences: for example:
    on the other hand, do support both of these.  Zero width ports are omitted (printed as
    comments) when generating verilog.
 
+### NonLocalAnchor
+The NonLocalAnchor operation (`hw.nla`) can be used to identify the unique
+ instance path of an operation globally.
+`hw.nla` can be used to attach nonlocal annotations and also for metadata
+ emission. 
+`hw.nla` defines a symbol and contains a list of module symbols followed
+ by a list of instance names corresponding to each module.
+ For example, in the following example, `@nla_0` specifies instance
+ "baz" in module `@FooNL`, followed by instance
+ `"bar"` in module `@BazNL`, followed by 
+ the wire named `"w"` in module `@BarNL`.
+
+`hw.nla` can define a unique instance path, and each element along the way
+ carries an annotation with class `circt. nonlocal`, which has a matching
+ `circt. nonlocal` field pointing to the global op. 
+ Thus instances participating in nonlocal paths are readily apparent.
+
+ In the following example the `@nla_0` is used in the verbatim op
+ to capture the final Verilog name of the wire `w`.
+
+The second `nla`, `@nla_1` doesn't specify the instance path and is anchored
+ on all instances of `@BarNL`.
+This example shows that the `hw.nla` can be used to attach a symbol with
+ any operation.
+
+
+``` mlir
+ firrtl.circuit "FooNL"   {
```

Comment:
please show how this works in non-firrtl too, since, for instance, we don't have annotations there.